python/jupytergis_lab/jupytergis_lab/notebook/gis_document.py

```diff
@@ -127,7 +131,8 @@ def add_geojson_layer(
             raise ValueError("Cannot set GeoJSON layer data and path at the same time")
 
         if path is not None:
-            parameters = {"path": path}
+            with open(path, "r") as fobj:
+                parameters = {"data": json.loads(fobj.read())}
```

Comment:
So the data itself is in the model now?

Reply:
In the case of the Notebook API yes

Reply:
BTW I don't think we should call it the Notebook API, but rather the Python API?

Reply:
Well, it only works if ran into a notebook 😓

Reply:
That's a limitation of the Python API, but a notebook has no API :smile: